apps/epic-web/src/inngest/functions/progress/lesson-completed.tsx

```diff
@@ -0,0 +1,247 @@
+import {inngest} from 'inngest/inngest.server'
+import {LESSON_COMPLETED_EVENT} from '@skillrecordings/skill-lesson/inngest/events'
+import {prisma} from '@skillrecordings/database'
+import {Redis} from '@upstash/redis'
+import LessonCompleteEmail from 'emails/lesson-complete-email'
+import {getModuleProgress} from '@skillrecordings/skill-lesson/lib/module-progress'
+import {sendTheEmail} from 'server/send-the-email'
+import {getLessonWithModule} from 'lib/lessons'
+import {
+  EMAIL_WRITING_REQUEST_COMPLETED_EVENT,
+  EMAIL_WRITING_REQUESTED_EVENT,
+} from 'inngest/events'
+
+const redis = new Redis({
+  url: process.env.UPSTASH_REDIS_REST_URL!,
+  token: process.env.UPSTASH_REDIS_REST_TOKEN!,
+})
+
+export const lessonCompleted = inngest.createFunction(
+  {
+    id: 'lesson-completed',
+    name: 'Lesson Completed',
+  },
+  {event: LESSON_COMPLETED_EVENT},
+  async ({event, step}) => {
+    const UNSUBSCRIBED_KEY = `unsubscribed:${event.user.id}:kody-the-encouragement-bot`
+
+    const isUnsubscribed = await step.run(
+      'Check if Learner is Unsubscribed',
+      async () => {
+        return await redis.get(UNSUBSCRIBED_KEY)
+      },
+    )
+
+    if (isUnsubscribed) {
+      return 'learner has opted out of lesson progress notifications'
+    }
+
+    const lessonWithModule = await step.run(
+      'Get Lesson With Module',
+      async () => {
+        return await getLessonWithModule(event.data.lessonSanityId as string)
+      },
+    )
+
+    const FIRST_LESSON_KEY = `first-lesson:${event.user.id}:${lessonWithModule.module.slug.current}`
+
+    const hasReceivedFirstLessonEmail = await step.run(
+      'Check if Received First Lesson Email',
+      async () => {
+        return await redis.get(FIRST_LESSON_KEY)
+      },
+    )
```

Comment:
we are only gonna send an email when they complete the first lesson in the module for now and might expand or add some other varieties later